tools/StaticAnalysis/BreakingChangeAnalyzer/BreakingChangeAnalyzer.cs

```diff
@@ -0,0 +1,968 @@
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Serialize the cmdlets so they can be compared to change modules later
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <param name="cmdlets"></param>
+        private void SerializeCmdlets(string fileName, List<CmdletBreakingChangeMetadata> cmdlets)
+        {
+            string json = JsonConvert.SerializeObject(cmdlets, Formatting.Indented);
+            File.WriteAllText(fileName, json);
+        }
+
+        /// <summary>
+        /// Deserialize the cmdlets to compare them to the changed modules
+        /// </summary>
+        /// <param name="fileName"></param>
+        /// <returns></returns>
+        private List<CmdletBreakingChangeMetadata> DeserializeCmdlets(string fileName)
+        {
+           return JsonConvert.DeserializeObject<List<CmdletBreakingChangeMetadata>>(File.ReadAllText(fileName));
+        }
+
+        /// <summary>
+        /// Run all of the different breaking change checks that we have for the tool
+        /// </summary>
+        /// <param name="oldCmdlets"></param>
+        /// <param name="newCmdlets"></param>
+        /// <param name="issueLogger"></param>
+        private void RunBreakingChangeChecks(
+            IList<CmdletBreakingChangeMetadata> oldCmdlets,
+            IList<CmdletBreakingChangeMetadata> newCmdlets,
+            ReportLogger<BreakingChangeIssue> issueLogger)
+        {
+            // Create a dictionary that maps a cmdlet name (and alias) to the corresponding metadata
+            IDictionary<string, CmdletBreakingChangeMetadata> cmdletMap = 
+                new Dictionary<string, CmdletBreakingChangeMetadata>();
+
+            // For each cmdlet in the new assembly, add it to the dictionary
+            foreach (var newCmdlet in newCmdlets)
+            {
+                cmdletMap.Add(newCmdlet.Name, newCmdlet);
+
+                foreach (var alias in newCmdlet.AliasList)
+                {
+                    if (!cmdletMap.ContainsKey(alias))
+                    {
+                        cmdletMap.Add(alias, newCmdlet);
+                    }
+                }
+            }
+
+            // For each cmdlet in the old assembly, check to see if it's in the new
+            // assembly, and if so, do the necessary breaking change checks
+            foreach (var oldCmdlet in oldCmdlets)
+            {
+                // If the cmdlet is in the new assembly, carry out the breaking change checks
+                if (cmdletMap.ContainsKey(oldCmdlet.Name))
+                {
+                    // Get the cmdlet metadata from the new assembly
+                    var newCmdlet = cmdletMap[oldCmdlet.Name];
+
+                    CheckForRemovedCmdletAlias(oldCmdlet, newCmdlet, issueLogger);
```

Comment:
This feels like too many methods in one class and that the methods aretoo large.  I would consider refactoring the methods so that they are extensions over the metadata types.  So you could have CmdletMetadata.CheckForBreakingChanges, which took the old version as an input.  This leaves the metdata types as poco types, but allows you to access their private data, and encapsulates the logic that knows about parameters in the parameter metadata